### PR TITLE
[Main Course] - Mining

### PR DESCRIPTION
**Pull request by jz263**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+上一期区块链介绍文章发布后，不少小伙伴表示对挖矿比较感兴趣，刚好前段时间我也试过挖矿究竟能产生多少收益。为了满足大家的好奇心，今天我想挖以太币为例，手把手教大家如何利用手边的显卡挖矿。
+
+## 挖矿真的有收益吗？
+想必这是大家最关心的一个问题，先说结论：如果你有一个原价买到的不错的显卡(英伟达16系以上)，是有一定收益的。按照目前的以太币价格（2000-3000美元/个），并且尽可能不让显卡在过高温度下长时间运行的话，去掉电费后大致能在6个月左右收回显卡的成本。虽然现在显卡价格还是远高于市场价，但是不是也可以考虑"打不过就加入"，在几个月后凭实力升级自己的显卡呢？
+
+如果你心动或者感兴趣了，那么请继续往下阅读吧：
+
+## 材料准备：
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:8]: 可以加下小标题
## 1

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## 步骤教学
+### 第一步：下载挖矿程序
+我选择的挖矿程序是pheonix miner，该软件号称有全网最低的抽成比例(0.65%)，亲测运行也较为稳定。
+目前有三种下载方式，如果你能够科学上网，可以尝试前两种方式：
+方式1：官网下载http://phoenixminer.org/
+官网上能点的基本上都是download按钮，只要确保下载的是对应计算机系统的版本即可
+
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-20/1624201267936-1624201264(1).png)
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:29]: 可以用下以前post的图片格式

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-20/1624202098120-image.png)
+
+注册成功后在钱包中点击钱包总览 - 充值 - 数字货币充值，选择币种为**以太坊**，选好后复制右方地址，该地址即为币安帮你保管的以太坊钱包地址：
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-20/1624203043450-image.png)
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-20/1624202944558-image.png)
+
+第三步：选择矿池
+这里推荐在https://ethermine.org/里选择以太币矿池。
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:59]: 显示上没有换行 行尾要在两个空格

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+第四步：配置挖矿脚本
+
+回到第一步下载的Pheonix Miner 文件夹下，会发现有一堆.bat后缀的脚本，这些是预先配置好的链接不同矿池的脚本代码，我们并不需要，只需要重新写一个新的脚本即可。创建一个新的txt文件，命名可以为：0_start_miner
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-20/1624204043440-image.png)
+
+打开后复制下列代码：
+
+```
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:75]: 看看有没有相应的高亮
```python这种的

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+REM setx GPU_MAX_HEAP_SIZE 100
+REM setx GPU_USE_SYNC_OBJECTS 1
+REM setx GPU_MAX_ALLOC_PERCENT 100
+REM setx GPU_SINGLE_ALLOC_PERCENT 100
+
+REM IMPORTANT: Replace the ETH address with your own ETH wallet address in the -wal option (Rig001 is the name of the rig)
+PhoenixMiner.exe -pool ssl://asia1.ethermine.org:5555 -pool2 ssl://us1.ethermine.org:5555 -wal 你的以太坊钱包地址.Rig001 -cdmrs -ttli 66
+pause
+```
+其中最关键的是最后一行命令行，**一定记得把之前复制的的以太坊钱包地址拷贝到命令行对应位置**，否则你的账户上收不到矿池给你发放的"工资"。此外-pool以后配置的是矿池的服务器地址和端口，这里应该和我们在第三步矿池网站中看到的服务器参数一致，-cdmrs -ttli 66表示显卡温度到达66摄氏度是会适当降低算力以保护下显卡。更多的命令行配置参数可以在docs - arguments里面找到，感兴趣的小伙伴可以研究一下，如果有有趣的配置方式不妨在评论区分享一下。
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:90]: 显卡温度最好再强调一下，算是一个风险点

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+配置完成后，将txt文件后缀改成.bat,双击运行后命令行的log大致会如下所示：
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-21/1624204861169-image.png)
+
+第五步：查看运行状况
+回到https://ethermine.org/网站，这时在miner address中输入你刚刚配置在命令行里的以太坊钱包地址，就可以实时监测到你的显卡产出情况了：
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:97]: .org/后面加个空格，不然链接会有点问题

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+![](https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-6-21/1624205203624-1624205114(1).png)
+
+最后需要注意的是，获得的收益不是立即从矿池进入你的钱包账户中，而是先暂存在矿池里，在unpaid balance处显示出来，在每月28号，或得到一定数量的以太币(一般是0.01个，在settings中可以看到具体发放规则)，才会一起发放到钱包账户中。
+
+收益：
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:103]: 换行

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+我的显卡型号是Nvidia 2070s，在显卡不过热的前提下，真实算力大概为34MH/s，一天跑满24小时的收益在0.0013个以太币左右，按照目前的价格(2000-2500$/个)，每天的收益在16-20元之间。电费大概需要6-8度/天，电费成本大致是收益的1/6。综合下来，差不多算是每天一杯manner吧。
+总的来说，挖矿的收益短期来看并不显著，加上虚拟货币价格波动很大，如果为了挖矿高价购买或囤积显卡，回本周期会很长且风险较高，几乎没有投资的必要。但如果是手边刚好有空置的显卡，单纯为了实现每天一杯咖啡自由，全托管式的挖矿不失为一种选择。
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:105]: 能不能再来一个简短的总结什么的

File: docs/Main_Course/One_Day_Build/2021-06-21-Mining.md (hunk 2 above)
Comment (BulletTech2021) [reply docs/Main_Course/One_Day_Build/2021-06-21-Mining.md:29]:
```<figure>
  <img src="https://dummyimage.com/600x400/eee/aaa" width="300" />
  <figcaption>Image caption</figcaption>
</figure>
```

**Comment by jhqian0909** [comment]
LGTM

**Review by jhqian0909 — APPROVED** [review]

**Review by BulletTech2021 — APPROVED** [review]
图片大小要调整，其他没问题

**Pull request merged**